src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -35,7 +35,8 @@ public class HubConnection
         private readonly object _pendingCallsLock = new object();
         private readonly CancellationTokenSource _connectionActive = new CancellationTokenSource();
         private readonly Dictionary<string, InvocationRequest> _pendingCalls = new Dictionary<string, InvocationRequest>();
-        private readonly ConcurrentDictionary<string, InvocationHandler> _handlers = new ConcurrentDictionary<string, InvocationHandler>();
+        private readonly ConcurrentDictionary<string, List<InvocationHandler>> _handlers = new ConcurrentDictionary<string, List<InvocationHandler>>();
+        private object _handlerLock = new object();
```

Comment:
What is this for?

Reply:
For safely copying and removing callbacks in dispose

Reply:
The lock should be per list not per connection.

Reply:
Just use the list as the lock.